core/src/main/java/org/mule/runtime/core/internal/processor/AsyncDelegateMessageProcessor.java

```diff
@@ -318,67 +317,61 @@ public void setMaxConcurrency(Integer maxConcurrency) {
-    private final SchedulerService schedulerService;
-    private final Supplier<SchedulerConfig> schedulerConfigSupplier;
-    private final String location;
     private final Consumer<CoreEvent> eventDispatcher;
 
-    private Scheduler queueDispatcherScheduler;
-    private volatile ScheduledFuture<?> scheduledDrain;
+    private final LazyValue<Scheduler> queueDispatcherScheduler;
 
     public QueueBackpressureHandler(SchedulerService schedulerService, Supplier<SchedulerConfig> schedulerConfigSupplier,
                                     Consumer<CoreEvent> eventDispatcher, String location) {
-      this.schedulerService = schedulerService;
-      this.schedulerConfigSupplier = schedulerConfigSupplier;
-
       this.asyncQueue = new LinkedBlockingQueue<>();
       this.eventDispatcher = eventDispatcher;
 
-      this.location = location;
+      this.queueDispatcherScheduler = new LazyValue(() -> {
+        final SchedulerConfig schedulerConfig = schedulerConfigSupplier.get().withName(location + " - queue dispatcher")
+            .withMaxConcurrentTasks(1);
+        return schedulerService.customScheduler(schedulerConfig);
+      });
     }
 
     public void handleBackpressure(CoreEvent event) {
       asyncQueue.offer(event);
-    }
 
-    public void asyncTaskFinished() {
-      if (scheduledDrain == null) {
-        synchronized (this) {
-          if (scheduledDrain == null) {
-            scheduledDrain = queueDispatcherScheduler.schedule(() -> {
-              while (asyncQueue.size() > 0) {
-                try {
-                  final CoreEvent event = asyncQueue.peek();
-                  if (event != null) {
-                    eventDispatcher.accept(event);
-                    asyncQueue.remove(event);
-                  }
-                } catch (FromFlowRejectedExecutionException free) {
-                  synchronized (this) {
-                    scheduledDrain = null;
-                  }
-                  // nothing to do, a retry will come next
-                  return;
-                }
+      synchronized (asyncQueue) {
+        asyncQueue.notify();
+      }
+
+      queueDispatcherScheduler.get().execute(() -> {
+        while (!currentThread().isInterrupted()) {
+          try {
+            final CoreEvent queuedEvent = asyncQueue.peek();
+            if (queuedEvent != null) {
+              eventDispatcher.accept(queuedEvent);
+              asyncQueue.remove(queuedEvent);
+            } else {
+              synchronized (asyncQueue) {
+                asyncQueue.wait();
               }
-            }, 200, MILLISECONDS);
+            }
+          } catch (FromFlowRejectedExecutionException free) {
+            // Nothing to do, let next iteration catch it.
+            yield();
+          } catch (InterruptedException e) {
+            currentThread().interrupt();
+            return;
           }
         }
-      }
+      });
     }
 
     @Override
     public void start() {
```

Comment:
delete method then. Doesn't look like you're actually forced to implement Startable